packages/tempus/src/Thyra_ImplicitAdjointModelEvaluator.hpp

```diff
@@ -0,0 +1,119 @@
+// @HEADER
+// ****************************************************************************
```

Comment:
Is there a reason these adapters cannot be tested on the SinCos test case you are testing the new capabilities on?  I'm not requiring you to do this, just asking out of my own interest.  If it's possible, I can try it / add a test on the Piro side.

Reply:
The ImplicitAdjointModelEvaluator is tested in almost every adjoint sensitivity test, since they all just supply a single ME (in which case Tempus creates this ImplicitAdjointModelEvaluator for the 2nd one).  I added an adjoint ME for the SinCos model and modified one adjoint sensitivity test to use it explicitly.  So both paths are tested.

Reply:
Ah I see.  Makes sense, thanks for clarifying.